### PR TITLE
The join exercise is done.

### PR DESCRIPTION
**Pull request by leeKol**
Hi, I'd love for your review.

### CODE REVIEW

File: join.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import itertools
+
+
+def join(*lists, sep='-'):
+    """
+    The function receives an unlimited number of lists and another parameter called sep (optional)
+    and returns one list consisting of all the lists received as parameters. If the sep parameter is provided,
```
Comment (itayday) [inline join.py]: Notice that there is an indentation problem in the documentation, please fix it.

File: join.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import itertools
+
+
+def join(*lists, sep='-'):
```
Comment (talbarfm) [inline join.py]: Would write separator, and not sep

File: join.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    and returns one list consisting of all the lists received as parameters. If the sep parameter is provided,
+    it threads it as an object between any two lists, and if it is not provided, it threads the character "-" instead.
+    :param lists: The lists to be merged into one list.
+    :param sep: The separator between the lists.
+    :return: The merged list.
+    """
+    if not lists:
+        return None
+    res_lst = []
+    for lst in lists:
```
Comment (talbarfm) [inline join.py]: read about the built in function zip, or the function itertools.chain and reduce this redundant for loop

File: join.py (hunk 2 above)
Comment (talbarfm) [inline join.py]: Add return types and type hints - apply in all pull requests 
https://docs.python.org/3/library/typing.html
Read about it here

File: join.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    """
+    The function receives an unlimited number of lists and another parameter called sep (optional)
+    and returns one list consisting of all the lists received as parameters. If the sep parameter is provided,
+    it threads it as an object between any two lists, and if it is not provided, it threads the character "-" instead.
+    :param lists: The lists to be merged into one list.
+    :param sep: The separator between the lists.
+    :return: The merged list.
+    """
+    if not lists:
+        return None
```
Comment (talbarfm) [inline join.py:15]: This behavior should be documented, I would return an empty list and not None  - will be easier to handle when other functions use it.
Read this - 
https://betterprogramming.pub/please-stop-returning-none-from-python-functions-ff98e492b4b0
And think about what the outputs of these inputs will be - 
[]
[[],[],[],]

File: join.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    res_lst = []
+    for lst in lists:
+        res_lst += list(itertools.chain(lst))
```
Comment (talbarfm) [inline join.py]: you are using itetools.chain incorrectly - the idea is to give it multiple lists, not a single list . 
If im not mistaken - what you are doing is creating a new list with the same element of lst, which is redundant.

**Review by talbarfm — CHANGES_REQUESTED** [review]

File: join.py (hunk 4 above)
Comment (leeKol) [reply join.py:15]: Hi,
The requirements of the exercise are as follows:
הקריאה join() תחזיר None או שגיאה, לבחירתכם
So I returned None.

**Review by lovodkin93 — APPROVED** [review]
well done!

**Pull request merged**